### PR TITLE
adds post telemetry

### PR DESCRIPTION
**Pull request by jonathanbarton**
This is the update to allow the sat modem to hit our defined callback URL, and post its data to the database for next retrieval from the clients. Without overengineering (I hope), set the stage for a couple possibilities: 
-- minimizing the payload from the sat modem to streamline costs vs readability of the data packet
-- allowing for additional sensors/data formats for sensors to be modified on the server and require less changes to the code on the raspberry pi (hopefully minimizing introduction of bugs)
-- allowing for us to expand to other device types besides the RockBlock satellite modem in the near future.

The data path works like this:
1. Sat modem sends data packet to satellite, satellite reformats into a POST request to a specified callback URL (current device works this way)
2. Our callback URL's endpoint creates a custom parser (currently we only have one) that uses a mapping model that maps the fields from the POST body to the attributes in our mongodb mongoose model(s). 
3. A transform module does some _pre-handling of the data if needed (i.e. converting the data field from hex encoding to ascii and then to a JSON object), and then walks through each of the mapped fields and their individual transform functions (i.e. if we need to convert decimal format of Lat/Long coords to degrees, we can do it here)
4. The parser takes this transformed data object and pipes it into the related Mongoose model, returns it to the controller for saving and the http success/error response.

1.

### CODE REVIEW

File: src/server/services/parsers/rockblock.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if (hasTransform && hasField) {
+      const transformFunctionName = Mappings[telemetryAttribute].transform;
+      const transform = Transforms[transformFunctionName];
+      const targetName = Mappings[telemetryAttribute].field;
+      const target = body[targetName];
+      acc[telemetryAttribute] = transform(target);
+    } else {
+      if (hasField) {
+        acc[telemetryAttribute] = Mappings[telemetryAttribute].field;
+      }
```
Comment (jonathanbarton) [inline src/server/services/parsers/rockblock.js]: i think this reduce method will fail cuz im not returning the accumulator. time to write a few more tests

**Comment by jonathanbarton** [comment]
@morrissinger @nehaabrol87 review this week when you get a chance

**Comment by nehaabrol87** [comment]
Looks solid :1st_place_medal:

**Comment by nehaabrol87** [comment]
ooops sorry merge conflicts :(

**Pull request merged**